

<launch>
    <arg name="bag"                 default="husky_outdoor4" />
    <arg name="source_path"         default="/home/wl/Desktop/husky" />
    <arg name="target_path"         default="/home/wl/Desktop/husky" />
    <arg name="source_bag_path"     default="$(arg source_path)/$(arg bag).bag" />
    <arg name="target_bag_path"     default="$(arg target_path)/$(arg bag).bag" />
    <arg name="time_start"          default="-1" />
    <arg name="time_end"            default="-1" />
    <arg name="vicon_add"           default="true" />
    <arg name="vicon_topic_in"      default="/optitrack" />
    <arg name="vicon_topic_out"     default="/optitrack" />
    <arg name="lidar_add"           default="true" />
    <arg name="lidar_topic_in"      default="/ouster/points" />
    <arg name="lidar_topic_out"     default="/ouster/points" />
    <arg name="gps_add"             default="true" />
    <arg name="gps_topic_in"        default="/gps/fix" />
    <arg name="gps_topic_out"       default="/gps/fix" />
    <arg name="rtk_add"             default="true" />
    <arg name="rtk_topic_in"        default="/fix" />
    <arg name="rtk_topic_out"       default="/fix" />
    <arg name="camera_add"          default="true" />
    <arg name="camera_topic_in"     default="/t265/fisheye1/image_raw/compressed, /t265/fisheye2/image_raw/compressed" />
    <arg name="camera_topic_out"    default="/t265/fisheye1/image_raw, /t265/fisheye2/image_raw" />
    <arg name="imu_add"             default="true" />
    <arg name="imu_topic_in"        default="/t265/imu" />
    <arg name="imu_topic_out"       default="/t265/imu" />
    <arg name="wheel1_add"          default="true" />
    <arg name="wheel1_topic_in"     default="/joint_states" />
    <arg name="wheel1_topic_out"    default="/joint_states" />
    <arg name="wheel2_add"          default="true" />
    <arg name="wheel2_topic_in"     default="/husky_velocity_controller/odom" />
    <arg name="wheel2_topic_out"    default="/husky_velocity_controller/odom" />


    <!-- MASTER NODE! -->
    <node name="rosbag_editor" pkg="rosbag_editor" type="rosbag_editor" output="screen" required="true">
<!--    <node name="rosbag_editor" pkg="rosbag_editor" type="rosbag_editor" output="screen" required="true" launch-prefix="gdb -ex run &#45;&#45;args">-->
        <param name="source_bag_path"   type="str"      value="$(arg source_bag_path)" />
        <param name="target_bag_path"   type="str"      value="$(arg target_bag_path)" />
        <param name="time_start"        type="double"   value="$(arg time_start)" />
        <param name="time_end"          type="double"   value="$(arg time_end)" />
        <param name="vicon_add"         type="bool"     value="$(arg vicon_add)" />
        <param name="lidar_add"         type="bool"     value="$(arg lidar_add)" />
        <param name="gps_add"           type="bool"     value="$(arg gps_add)" />
        <param name="rtk_add"           type="bool"     value="$(arg rtk_add)" />
        <param name="camera_add"        type="bool"     value="$(arg camera_add)" />
        <param name="imu_add"           type="bool"     value="$(arg imu_add)" />
        <param name="wheel1_add"        type="bool"     value="$(arg wheel1_add)" />
        <param name="wheel2_add"        type="bool"     value="$(arg wheel2_add)" />
        <param name="vicon_topic_in"    type="str"      value="$(arg vicon_topic_in)" />
        <param name="vicon_topic_out"   type="str"      value="$(arg vicon_topic_out)" />
        <param name="lidar_topic_in"    type="str"      value="$(arg lidar_topic_in)" />
        <param name="lidar_topic_out"   type="str"      value="$(arg lidar_topic_out)" />
        <param name="gps_topic_in"      type="str"      value="$(arg gps_topic_in)" />
        <param name="gps_topic_out"     type="str"      value="$(arg gps_topic_out)" />
        <param name="rtk_topic_in"      type="str"      value="$(arg rtk_topic_in)" />
        <param name="rtk_topic_out"     type="str"      value="$(arg rtk_topic_out)" />
        <param name="camera_topic_in"   type="str"      value="$(arg camera_topic_in)" />
        <param name="camera_topic_out"  type="str"      value="$(arg camera_topic_out)" />
        <param name="imu_topic_in"      type="str"      value="$(arg imu_topic_in)" />
        <param name="imu_topic_out"     type="str"      value="$(arg imu_topic_out)" />
        <param name="wheel1_topic_in"   type="str"      value="$(arg wheel1_topic_in)" />
        <param name="wheel1_topic_out"  type="str"      value="$(arg wheel1_topic_out)" />
        <param name="wheel2_topic_in"   type="str"      value="$(arg wheel2_topic_in)" />
        <param name="wheel2_topic_out"  type="str"      value="$(arg wheel2_topic_out)" />
    </node>
</launch>
